### PR TITLE
docs: add page `instantiating-wallets`

### PR DESCRIPTION
**Pull request by Torres-ssf**
This doc page is part of the [improvements](https://github.com/FuelLabs/fuels-ts/issues/1919) on the Wallet docs section.

This guide covers how to instantiate wallets and how to connect them to a `Provider`

### CODE REVIEW

**Comment by petertonysmith94** [comment]
@Torres-ssf this requires a changeset

File: apps/docs/src/guide/wallets/instantiating-wallets.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+::: code-group
+
+```ts [Bech32]
+const address =
+  "fuel14kjrdcdcp7z4l9xk0pm3cwz9qnjxxd04wx4zgnc3kknslclxzezqyeux5d";
+
+const myWallet = Wallet.fromAddress(address);
+
+expect(myWallet).toBeDefined();
```
Comment (petertonysmith94) [inline apps/docs/src/guide/wallets/instantiating-wallets.md:47]:
```suggestion
```

Suggest removing `expect` to match the other snippets.

File: apps/docs/src/guide/wallets/instantiating-wallets.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+```ts [Hex]
+const address =
+  "0xada436e1b80f855f94d678771c384504e46335f571aa244f11b5a70fe3e61644";
+
+const myWallet = Wallet.fromAddress(address);
+
+expect(myWallet).toBeDefined();
```
Comment (petertonysmith94) [inline apps/docs/src/guide/wallets/instantiating-wallets.md:56]:
```suggestion
```

Suggest removing `expect` to match the other snippets.

**Review by petertonysmith94 — DISMISSED** [review]

File: apps/docs/src/guide/wallets/instantiating-wallets.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+From a JSON wallet:
+
+<<< @/../../docs-snippets/src/guide/wallets/instantiating-wallets.test.ts#instantiating-wallets-6{ts:line-numbers}
+
+## Instantiating Locked Wallets
+
+You can also instantiate [`WalletLocked`](../../api/Account/WalletLocked) instances using just the wallet address:
+
+::: code-group
```
Comment (danielbate) [inline apps/docs/src/guide/wallets/instantiating-wallets.md:39]: Why code group here and not a test snippet?

Comment (petertonysmith94) [reply apps/docs/src/guide/wallets/instantiating-wallets.md:39]: I believe it's to show that a Wallet can be instantiated from different address formats, in this case  `Bech32` and `B255`.

Comment (arboleya) [reply apps/docs/src/guide/wallets/instantiating-wallets.md:39]: We can still use a code snippet inside the group.

Comment (Torres-ssf) [reply apps/docs/src/guide/wallets/instantiating-wallets.md]: I've decided to keep it simple and use just one example

**Review by petertonysmith94 — APPROVED** [review]

File: apps/docs/src/guide/wallets/instantiating-wallets.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+From a Hierarchical Deterministic (HD) derived key:
+
+<<< @/../../docs-snippets/src/guide/wallets/instantiating-wallets.test.ts#instantiating-wallets-5{ts:line-numbers}
+
+From a JSON wallet:
+
+<<< @/../../docs-snippets/src/guide/wallets/instantiating-wallets.test.ts#instantiating-wallets-6{ts:line-numbers}
+
+## Instantiating Locked Wallets
```
[the remark this answers is not in the document]
Comment (nedsalk) [reply apps/docs/src/guide/wallets/instantiating-wallets.md:35]: Resolved in https://github.com/FuelLabs/fuels-ts/pull/1947#discussion_r1543251737

**Review by nedsalk — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Pull request merged**